### PR TITLE
e2e: Test cases for the etcdctlv3 user commands.

### PR DESCRIPTION
**Pull request by ajityagaty**
New test cases have been added to test the "user" sub-commands of
the etcdctlv3 utility.

### CODE REVIEW

**Comment by ajityagaty** [comment]
@gyuho I have tried to follow the existing patterns used in other test cases. Please let me know if anything needs to be changed.

Also, I noticed that sometimes call to `proc.Expect` fails with an EOF (broken pipe). But if I try to run them individually the test cases pass. Have you seen this problem before?

**Comment by gyuho** [comment]
Thanks! And could you move all `user` tests to `ctl_v3_user_test.go`?

File: e2e/ctl_v3_user_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	stdIn       []string
+}
+
+func userAddTest(cx ctlCtx) {
+	defer close(cx.errc)
+
+	cmdSet := map[string]userCmdDesc{
+		// Adds a user name.
+		"add1": {
+			args:        []string{"add", "username", "--interactive=false"},
```
Comment (gyuho) [inline e2e/ctl_v3_user_test.go]: I think `--interactive=false` is not needed because the default is true.

File: e2e/ctl_v3_user_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+		"add": {
+			args:        []string{"add", "username", "--interactive=false"},
+			expectedStr: "User username created",
+			stdIn:       []string{"password"},
+		},
+		// Deletes the user name just added.
+		"delete1": {
+			args:        []string{"delete", "username"},
+			expectedStr: "User username deleted",
+			stdIn:       nil,
```
Comment (gyuho) [inline e2e/ctl_v3_user_test.go]: `stdIn:       nil` is not needed because the default is `nil`.

File: e2e/ctl_v3_user_test.go (hunk 1 above)
Comment (ajityagaty) [inline e2e/ctl_v3_user_test.go]: I noticed that for `user add` sub-command the default is true for `--interactive` flag. Hence I explicitly set it to false.

File: e2e/ctl_v3_user_test.go (hunk 2 above)
Comment (ajityagaty) [inline e2e/ctl_v3_user_test.go]: Will take care of it.

**Comment by ajityagaty** [comment]
@gyuho 
`FAIL: TestCtlV3UserPasswdNoTLS (1.11s)
	ctl_v3_user_test.go:122: userPasswdTest passwd: ctlV3User error (read /dev/ptmx: input/output error)`

The above test case passes sometimes and sometimes it fails. Could you please let me know if I have used the `expect` methods correctly? Or is there some kind of a race?

**Comment by gyuho** [comment]
@ajityagaty Go map is un-ordered, so you need to set test cases of `struct` slice or any other ordered data structure. Think that's why the test was failing?

File: e2e/ctl_v3_user_test.go (hunk 1 above)
Comment (gyuho) [inline e2e/ctl_v3_user_test.go]: You are right. nvm.

**Comment by ajityagaty** [comment]
@gyuho Thanks for pointing out about the map property. I didn't know about it. Will fix it and give it a try.

File: e2e/ctl_v3_user_test.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+import "testing"
+
+func TestCtlV3UserAdd(t *testing.T)          { testCtl(t, userAddTest) }
+func TestCtlV3UserAddNoTLS(t *testing.T)     { testCtl(t, userAddTest, withCfg(configNoTLS)) }
+func TestCtlV3UserAddClientTLS(t *testing.T) { testCtl(t, userAddTest, withCfg(configClientTLS)) }
+func TestCtlV3UserAddPeerTLS(t *testing.T)   { testCtl(t, userAddTest, withCfg(configPeerTLS)) }
+func TestCtlV3UserAddTimeout(t *testing.T)   { testCtl(t, userAddTest, withDialTimeout(0)) }
+
+func TestCtlV3UserDelete(t *testing.T)          { testCtl(t, userDelTest) }
```
Comment (gyuho) [inline e2e/ctl_v3_user_test.go]: let's keep only `TestCtlV3UserDelete` and `TestCtlV3UserPasswd` and delete `cx.dialTimeout > 0 && !isGRPCTimedout(err)` from `userDelTest` and `userPasswdTest`. Then LGTM. Thanks!

Comment (gyuho) [inline e2e/ctl_v3_user_test.go]: I meant keep everything with `TestCtlV3UserAdd*`. And delete variations in other commands.

**Comment by gyuho** [comment]
lgtm. Thanks!

**Pull request merged**